scripts/update_ambiguous_characters.py

```diff
@@ -0,0 +1,62 @@
+"""Generate the confusables.rs file from the VS Code ambiguous.json file."""
+import json
+import subprocess
+from pathlib import Path
+
+CONFUSABLES_RS_PATH = "crates/ruff/src/rules/ruff/rules/confusables.rs"
+AMBIGUOUS_JSON_URL = "https://raw.githubusercontent.com/hediet/vscode-unicode-data/main/out/ambiguous.json"
+
+
+def get_mapping_data() -> dict:
+    """
+    Get the ambiguous character mapping data from the vscode-unicode-data repository.
+
+    Uses the system's `curl` command to download the data,
+    instead of adding a dependency to a Python-native HTTP client.
+    """
+    content = subprocess.check_output(
+        ["curl", "-sSL", AMBIGUOUS_JSON_URL],
+        encoding="utf-8",
+    )
```

Comment:
I'm not very familiar with Python but would `urllib` be a viable (dependency free) option to get cross-platform support?

Reply:
I figured this is more bullet-proof and reasonably cross-platform, since cURL is easily available for all platforms. This script is only expected to be run (by a maintainer) when the upstream updates anyway, and I would assume said maintainer has cURL :)